src/main/java/com/conveyal/datatools/manager/models/transform/NormalizeFieldTransformation.java

```diff
@@ -0,0 +1,338 @@
+            int transformFieldIndex = getFieldIndex(fieldsFoundInZip, fieldName);
+
+            int modifiedRowCount = 0;
+            Long startMillis = System.currentTimeMillis();
+
+            // Write headers and processed CSV rows.
+            writer.write(headers);
+            while (csvReader.readRecord()) {
+                String originalValue = csvReader.get(transformFieldIndex);
+                String transformedValue = originalValue;
+
+                // Convert to title case, if requested.
+                if (capitalize) {
+                    transformedValue = convertToTitleCase(transformedValue);
+                }
+
+                // Perform substitutions, if requested.
+                if (performSubstitutions) {
+                    transformedValue = performSubstitutions(transformedValue);
+                }
+
+                // Re-assemble the CSV line and place in buffer.
+                String[] csvValues = csvReader.getValues();
+                csvValues[transformFieldIndex] = transformedValue;
+
+                // Write line to table (plus new line char).
+                writer.write(csvValues);
+
+                // Count number of CSV rows changed.
+                if (!originalValue.equals(transformedValue)) {
+                    modifiedRowCount++;
+                }
+            } // End of iteration over each row.
+            csvReader.close();
+            writer.flush();
+            Long stopMillis = System.currentTimeMillis();
+
+            // Copy csv input stream into the zip file, replacing the existing file.
+            try (
+                // Modify target zip file that we just read.
+                FileSystem targetZipFs = FileSystems.newFileSystem(tempZipPath, null);
+                // Stream for file copy operation.
+                InputStream inputStream =  new ByteArrayInputStream(stringWriter.toString().getBytes(StandardCharsets.UTF_8))
+            ) {
+                Path targetTxtFilePath = getTablePathInZip(tableName, targetZipFs);
+                Files.copy(inputStream, targetTxtFilePath, StandardCopyOption.REPLACE_EXISTING);
+                zipTarget.feedTransformResult.tableTransformResults.add(
+                    new TableTransformResult(tableName, 0, modifiedRowCount, 0)
+                );
+            }
+
+            // Replace original zip file with temporary working zip file.
+            // (This should also trigger a system IO update event, so subsequent IO calls pick up the correct file.
+            Files.move(tempZipPath, originalZipPath, StandardCopyOption.REPLACE_EXISTING);
+            Long transformEndMillis = System.currentTimeMillis();
+            LOG.info(
+                "Field normalization transformation successful: {} row(s) changed in {} ms ({} ms including file ops).",
+                modifiedRowCount,
+                stopMillis - startMillis,
+                transformEndMillis - transformStartMillis
+            );
```

Comment:
I think it's ok to remove this as the logging of the time it takes for a job to complete is already logged in the MonitorableJob#run method.

Reply:
Updated in https://github.com/ibi-group/datatools-server/pull/371/commits/c44c928b6bf4870e029e5856b29494bf57e47120.